massa-consensus-worker/src/state/prune.rs

```diff
@@ -332,6 +332,21 @@ impl ConsensusState {
         Ok(())
     }
 
+    fn prune_nonfinal_blocks_per_slot(&mut self) {
```

Comment:
This is implemented in a very heavy way. Here is a higher performance option:

```
self.nonfinal_active_blocks_per_slot.retain(|s, _| s.period > self.latest_final_blocks_periods[s.thread as usize]);
```

Reply:
Done.